extensions/ql-vscode/src/databases/local-databases-ui.ts

```diff
@@ -986,6 +997,104 @@ export class DatabaseUI extends DisposableObject {
     }
   }
 
+  /**
+   * Ask the user for a database directory. Returns the chosen database, or `undefined` if the
+   * operation was canceled.
+   */
+  private async chooseAndSetDatabase(
+    byFolder: boolean,
+    progress: ProgressCallback,
+  ): Promise<DatabaseItem | undefined> {
+    const uri = await chooseDatabaseDir(byFolder);
+    if (!uri) {
+      return undefined;
+    }
+
+    return await this.importDatabase(uri, byFolder, progress);
+  }
+
+  /**
+   * Ask the user for a parent directory that contains all databases.
+   * Returns all valid databases, or `undefined` if the operation was canceled.
+   */
+  private async chooseDatabasesParentFolder(
+    progress: ProgressCallback,
+  ): Promise<DatabaseItem[] | undefined> {
+    const uri = await chooseDatabaseDir(true);
+    if (!uri) {
+      return undefined;
+    }
+
+    const databases: DatabaseItem[] = [];
+    const failures: string[] = [];
+    const entries = await workspace.fs.readDirectory(uri);
+    const validFileTypes = [FileType.File, FileType.Directory];
+
+    for (const [index, entry] of entries.entries()) {
+      progress({
+        step: index + 1,
+        maxStep: entries.length,
+        message: `Importing ${entry[0]}`,
+      });
+
+      const subProgress: ProgressCallback = (p) => {
+        progress({
+          step: index + 1,
+          maxStep: entries.length,
+          message: `Importing ${entry[0]} (${p.step}/${p.maxStep}): ${p.message}`,
+        });
+      };
+
+      if (!validFileTypes.includes(entry[1])) {
+        void this.app.logger.log(
+          `Skip import ${entry}, invalid FileType: ${entry[1]}`,
+        );
+        continue;
+      }
+
+      try {
+        const databaseUri = Uri.joinPath(uri, entry[0]);
+        void this.app.logger.log(`Importing from ${databaseUri}`);
+
+        const database = await this.importDatabase(
+          databaseUri,
+          entry[1] === FileType.Directory,
+          subProgress,
+        );
+        if (database) {
+          databases.push(database);
+        } else {
+          failures.push(entry[0]);
+        }
+      } catch {
+        failures.push(entry[0]);
```

Comment:
The error message should be captured here and printed alongside the failures of each database. Something like this:

```suggestion
      } catch (e) {
        failures.push(`${entry[0]}: ${toErrorMessag(e)}`);
```

(I typed this by hand so the formatting is probably incorrect.)